### PR TITLE
Updated sql-plan-management.md

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added upgrade checklist for SPM

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6866 and https://github.com/pingcap/docs-cn/pull/6896
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-plan-management.md [hunk 1]
```diff
@@ -287,3 +287,41 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+  
+    select @@global.tidb_evolve_plan_baselines;
+  
+    -- If `tidb_evolve_plan_baselines` is still enabled, disable it. 
+  
+    set global tidb_evolve_plan_baselines = off;
+    ```
+
+* Before you upgrade from v4.0 to the current version, you must check whether the grammar of all the SQL statements corresponding to available SQL bindings is correct in the new version. If any grammatical errors, delete the corresponding SQL bindings. 
+Perform the following steps.
```
Comment (TomShawn) [inline sql-plan-management.md:310]:
```suggestion
* Before you upgrade from v4.0 to the current version, you must check whether the grammar of all the SQL statements corresponding to available SQL bindings is correct in the new version. If any grammatical errors, delete the corresponding SQL bindings. Perform the following steps.
```

**Comment by hfxsd** [comment]
/status/can-merge

File: sql-plan-management.md [hunk 2]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
     | `max_execution_time` | The longest duration for a query. |
 
 + `read_from_storage` is a special hint in that it specifies whether to read data from TiKV or from TiFlash when reading tables. Because TiDB provides isolation reads, when the isolation condition changes, this hint has a great influence on the evolved execution plan. Therefore, when this hint exists in the initially created binding, TiDB ignores all its evolved bindings.
+
+## Upgrade Checklist
```
Comment (TomShawn) [inline sql-plan-management.md:291]:
```suggestion
## Upgrade checklist
```

File: sql-plan-management.md [hunk 3]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+
+SQL Plan Management (SPM) may fail in upgrade due to compatibility issues. You need to check the environment before upgrade to make sure it succeeds.
```
Comment (TomShawn) [inline sql-plan-management.md:293]:
```suggestion
During cluster upgrade, SQL Plan Management (SPM) might cause compatibility issues and make the upgrade fail. To ensure a successful upgrade, you need to include the following list for upgrade precheck:
```

File: sql-plan-management.md [hunk 4]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
     | `max_execution_time` | The longest duration for a query. |
 
 + `read_from_storage` is a special hint in that it specifies whether to read data from TiKV or from TiFlash when reading tables. Because TiDB provides isolation reads, when the isolation condition changes, this hint has a great influence on the evolved execution plan. Therefore, when this hint exists in the initially created binding, TiDB ignores all its evolved bindings.
+
+## Upgrade Checklist
+
+SQL Plan Management (SPM) may fail in upgrade due to compatibility issues. You need to check the environment before upgrade to make sure it succeeds. 
+
+* When you upgrade from a version earlier than v5.2.0 (that is, v4.0, v5.0, and v5.1) to the current version, make sure `tidb_evolve_plan_baselines` is disabled. If not, you must disable it before upgrade. Perform the following steps.
```
Comment (TomShawn) [inline sql-plan-management.md:295]:
```suggestion
* When you upgrade from a version earlier than v5.2.0 (that is, v4.0, v5.0, and v5.1) to the current version, make sure that `tidb_evolve_plan_baselines` is disabled before the upgrade. To disable this variable, perform the following steps. 
```

File: sql-plan-management.md [hunk 5]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+    -- Check whether `tidb_evolve_plan_baselines` is disabled in the earlier version. 
+  
+    select @@global.tidb_evolve_plan_baselines;
+  
+    -- If `tidb_evolve_plan_baselines` is still enabled, disable it. 
+  
+    set global tidb_evolve_plan_baselines = off;
+    ```
+
+* Before you upgrade from v4.0 to the current version, you must check whether the grammar of all the SQL statements corresponding to available SQL bindings is correct in the new version. If any grammatical errors, delete the corresponding SQL bindings. Perform the following steps.
```
Comment (TomShawn) [inline sql-plan-management.md:309]:
```suggestion
* Before you upgrade from v4.0 to the current version, you need to check whether the syntax of all queries corresponding to the available SQL bindings is correct in the new version. If any syntax errors exist, delete the corresponding SQL binding. To do that, perform the following steps.
```

File: sql-plan-management.md [hunk 6]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    -- Check the SQL statements corresponding to available SQL bindings in the version to be upgraded.
```
Comment (TomShawn) [inline sql-plan-management.md:314]:
```suggestion
    -- Check the query corresponding to the available SQL binding in the version to be upgraded.
```

File: sql-plan-management.md [hunk 7]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+    select bind_sql from mysql.bind_info where source != 'builtin' and status = 'using';
+  
+    -- Verify the result from the above SQL query in the test environment of the new version. 
+  
+    bind_sql_0;
+    bind_sql_1;
+    ...
+  
+    -- In the case of a syntax error (ERROR 1064 (42000): You have an error in your SQL syntax), delete the corresponding binding. 
+    -- For any other errors (for example, tables are not found), it means the syntax is compatible. No intervention is needed.
```
Comment (TomShawn) [inline sql-plan-management.md:325]:
```suggestion
    -- For any other errors (for example, tables are not found), it means that the syntax is compatible. No other operation is needed. 
```

File: sql-plan-management.md (hunk 4 above)
Comment (Reminiscent) [inline sql-plan-management.md:295]:
```suggestion
* When you upgrade from a version earlier than v5.2.0 (that is, v4.0, v5.0, and v5.1) to the current version, make sure that the system variable `tidb_evolve_plan_baselines` is disabled before the upgrade. To disable this variable, perform the following steps.
```

File: sql-plan-management.md [hunk 8]
```diff
@@ -287,3 +287,40 @@ Because the baseline evolution automatically creates a new binding, when the que
...
+    ```
+
+* Before you upgrade from v4.0 to the current version, you must check whether the grammar of all the SQL statements corresponding to available SQL bindings is correct in the new version. If any grammatical errors, delete the corresponding SQL bindings. Perform the following steps.
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    -- Check the SQL statements corresponding to available SQL bindings in the version to be upgraded.
+  
+    select bind_sql from mysql.bind_info where source != 'builtin' and status = 'using';
```
Comment (Reminiscent) [inline sql-plan-management.md:316]:
```suggestion
    select bind_sql from mysql.bind_info where status = 'using';
```
I'll change the Chinese doc at the same time. It's a little mistake. It does not affect correctness, it just makes the SQL simpler.

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@Reminiscent PTAL again. Thanks!

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**